src/app/WriteHandler.cpp

```diff
@@ -634,6 +645,52 @@ Status WriteHandler::ProcessWriteRequest(System::PacketBufferHandle && aPayload,
     return status;
 }
 
+CHIP_ERROR WriteHandler::VerifyAttributeDataIBs(const TLV::TLVReader & aAttributeDataIBsReader, bool aIsGroupAction)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    TLV::TLVReader dataIBsReader(aAttributeDataIBsReader);
+
+    while (CHIP_NO_ERROR == (err = dataIBsReader.Next()))
+    {
+        chip::TLV::TLVReader dataReader;
+        AttributeDataIB::Parser element;
+        AttributePathIB::Parser attributePath;
+        ConcreteDataAttributePath dataAttributePath;
+        TLV::TLVReader reader = dataIBsReader;
+
+        SuccessOrExit(err = element.Init(reader));
+        SuccessOrExit(err = element.GetPath(&attributePath));
+
+        if (!aIsGroupAction)
+        {
+            SuccessOrExit(err = attributePath.GetEndpoint(&(dataAttributePath.mEndpointId)));
```

Comment:
Shouldn't we check `IsValidEndpointId`?

Reply:
Yes, added checking for IsValidEndpointId